tests/test_devices.py

```diff
@@ -255,6 +255,152 @@ def test_device_inventory_update_success(client, mocker):
     assert device._inventory == inventory
 
 
+class test_inputs:
+    def test_property_populated(client, mocker):
+        """Should check if the device property are correctly populated"""
+        device = AlarmDevice(client)
+        mocker.spy(device._connection, "query")
+        device.connect("username", "password")
+        inputs = {
+            0: "Entryway Sensor",
+            1: "Outdoor Sensor 1",
+            2: "Outdoor Sensor 2",
+        }
+        # Test
+        device.update()
+        device_inputs = {}
+        for input_id, name in device.inputs:
+            device_inputs.update({input_id: name})
+        assert device_inputs == inputs
+
+    def test_all_properties_empty(client, mocker):
+        """Should check store the e-connect System status in the inventory device object."""
+        device = AlarmDevice(client)
+        mocker.spy(device._connection, "query")
+        device.connect("username", "password")
+        # Test
+        device.update()
+        device._inventory = {}
+        device_inputs = list(device.inputs)
+        assert not device_inputs
```

Comment:
you should always test exactly what you get. In this case, you should get an empty dictionary, so `{}`. If you test `assert not`, this test succeeds if you return `None`, but that's not what you want as `None` will let the rest of the application crash (very likely).

Test `assert dict(device.inputs) == {}`